app/backend/src/clj/teet/thk/thk_import.clj

```diff
@@ -222,6 +229,111 @@
                                                     thk-mapping/activity-integration-info-fields)}))}))))}))]
      (task-updates rows))))
 
+(defn get-project-construction-activity-db-id
+  [db rows]
+  (let [construction-activity (first (filter (comp #{:activity.name/construction} :activity/name) rows))
+        construction-activity-integration-id (:activity-db-id construction-activity)
+        construction-activity-ids (lookup db [:integration/id construction-activity-integration-id])]
+    (:db/id construction-activity-ids)))
+
+(defn- get-project-attrs [db project-id rows]
+  (let [phase-est-starts (keep :thk.lifecycle/estimated-start-date rows)
+        phase-est-ends (keep :thk.lifecycle/estimated-end-date rows)
+        thk-project (lookup db [:thk.project/id project-id])]
+    {:prj (first rows)
+     :phases (group-by :thk.lifecycle/id rows)
+
+     :project-est-start (first (sort phase-est-starts))
+     :project-est-end (last (sort phase-est-ends))
+
+     :proj-integration-id (:integration/id thk-project)
+     :proj-db-id (:db/id thk-project)
+
+     :project-exists? (some? (:db/id thk-project))
+     :project-has-owner? (and (some? (:db/id thk-project))
+                           (project-db/project-has-owner? db [:thk.project/id project-id]))
+     :construction-activity-db-id (get-project-construction-activity-db-id db rows)}))
+
+
+(defn- get-task-type [db task-eid]
+  (ffirst (d/q '[:find (pull ?e [:task/type])
+                 :where
+                 [(missing? $ ?e :meta/deleted?)]
+                 :in $ ?e] db task-eid)))
+
+(defn- get-activity-tasks-eids [db activity-eid]
+  (let [query-result
+        (d/q '[:find (pull ?e [:activity/tasks])
+               :in $ ?e] db activity-eid)]
+    (when
+      (some? query-result)
+      (:activity/tasks (ffirst query-result)))))
+
+(defn- get-existing-task-db-id
+  "Search for the existing task with the same type and start and end dates"
+  [db construction-activity-id task-type start-date end-date]
+  (first (reduce
+           #(if
```

Comment:
Could be more readable with fn that had named args. also (identity x) is same as x.